backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportNamer.kt

```diff
@@ -331,7 +332,10 @@ internal class ObjCExportNamerImpl(
         assert(mapper.isObjCProperty(property))
 
         StringBuilder().apply {
-            append(property.name.asString())
+            val candidate = property.name.asString()
```

Comment:
There is an extension point to prevent this kind of name conflicts: you can override `reserved` method in `propertyNames`.